spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,284 @@
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'redirects to root path' do
+          expect(response).to redirect_to root_path
+        end
+
+        it 'shows error in flash message' do
+          expect(flash[:alert]).to match "Sorry, registration limit exceeded for #{conference_with_open_registration.title}"
+        end
+      end
+
+      context 'successful request' do
+        before do
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'assigns registration and user variables' do
+          expect(assigns(:registration)).to be_instance_of(Registration)
+          expect(assigns(:user)).to be_instance_of(User)
+        end
+
+        it 'renders the new template' do
+          expect(response).to render_template('new')
+        end
+      end
+    end
+
+    describe 'GET #show' do
+      before do
+        @registration = create(:registration, conference: conference_with_open_registration, user: user)
+      end
+
+      context 'user has purchased a ticket' do
+        before do
+          @ticket = create(:ticket, conference: conference_with_open_registration)
+          @purchased_ticket = create(:ticket_purchase, conference: conference_with_open_registration,
+                                                       user: user,
+                                                       ticket: @ticket)
+          get :show, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'renders the show template' do
+          expect(response).to render_template('show')
+        end
+
+        it 'assigns workshops variable' do
+          expect(assigns(:workshops)).to eq @registration.workshops
+        end
+
+        it 'assigns price of purchased tickets to total_price and purchased tickets to tickets' do
+          expect(assigns(:total_price)).to eq Money.new(10000, 'USD')
+          expect(assigns(:tickets)).to match_array [@purchased_ticket]
+        end
+      end
+
+      context 'user has not purchased any ticket' do
+        before do
+          get :show, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'assigns 0 dollars to total_price and empty array to tickets variables' do
+          expect(assigns(:total_price)).to eq Money.new(0, 'USD')
+          expect(assigns(:tickets)).to match_array []
+        end
+      end
+    end
+
+    describe 'GET #edit' do
+      before do
+        create(:registration, conference: conference_with_open_registration, user: user)
+        get :edit, conference_id: conference_with_open_registration.short_title
+      end
+
+      it 'renders the edit template' do
+        expect(response).to render_template('edit')
+      end
+    end
+
+    describe 'POST #create' do
+      context "tickets are available and user hasn't bought any" do
```

Comment:
The context and the tests in it don't match a lot. IMO the successful creation of a registration happens anyway, so it should not be tested inside this context.